pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -1208,10 +1212,10 @@ func TestValidatePipelineResults_Failure(t *testing.T) {
 	results := []PipelineResult{{
```

Comment:
Is the reason this is invalid because the result is not an object type result?

Reply:
Just to check my understanding, `tasks.a-task.results.output.output` is no longer invalid, as it will be treated as the "output" key on the "output" result, and if such a result doesn't exist, there will be an error? Are the cases where the result or property doesn't exist tested anywhere?

Reply:
> Just to check my understanding, `tasks.a-task.results.output.output` is no longer invalid, as it will be treated as the "output" key on the "output" result, and if such a result doesn't exist, there will be an error? Are the cases where the result or property doesn't exist tested anywhere?

Great questions!

> if such a result doesn't exist, there will be an error?
Yes, this result ref validation is in place on the [reconciler level](https://github.com/tektoncd/pipeline/blob/33b3eac346a37ae7943f210fa13d9c4ea7d571fa/pkg/reconciler/pipelinerun/resources/validate_dependencies.go#L60-L87) (not here in the admission webhook level). Specifically, the error for the case when a result name doesn't exist will be returned from line 84.

> Are the cases where the result or property doesn't exist tested anywhere?
Yes, there will definitely be a validation for that. I already added the `property` field in the [`ResultRef` struct](https://github.com/tektoncd/pipeline/pull/4878/files#diff-49d33732c3f59a974dc47badc9ea139fe28e0e9b059c1f0aa478da3f78deaa09R26-R30). So IIUC the validation for object result properties will be done on the reconciler level too (will be implemented in future pr).